Sources/SourceControl/RepositoryManager.swift

```diff
@@ -140,6 +122,9 @@ public class RepositoryManager {
     // specifier but then is capable of efficiently determining if two
     // repositories map to the same location.
     private var repositories: [String: RepositoryHandle] = [:]
+
+    /// Lock to protect concurrent reads and mutations to repositories registory.
+    private var repositoriesLock = Lock()
```

Comment:
Should we prefer a queue versus a lock? I like the queue pattern for serialized access to state, even though they are basically the same.